packages/compass-sidebar/src/components/sidebar/sidebar.jsx

```diff
@@ -133,11 +132,21 @@ class Sidebar extends PureComponent {
   }
 
   _calculateRowHeight({index}) {
-    const db = this.props.databases.databases[index];
-    let height = ROW_HEIGHT;
-    if (this.props.databases.expandedDblist[db._id]) {
-      height += db.collections.length * ROW_HEIGHT + EXPANDED_WHITESPACE;
+    const { filterRegex, databases, expandedDbList } = this.props.databases;
+    const defaultExpanded = Boolean(filterRegex);
+    const db = databases[index];
+    const collectionsLength = ['initial', 'fetching'].includes(
+      db.collectionsStatus
+    )
+      ? db.collection_count ?? 0
+      : db.collections.length;
+
+    let height = HEADER_ROW_HEIGHT;
+
+    if (expandedDbList[db._id] ?? defaultExpanded) {
```

Comment:
I don't really understand why db wouldn't have _id?

Reply:
OK sorry I misread that, then.

Reply:
The logic for this collapsing is not obvious at a glance at all, I'll try my best to have tests that can highlight the use-cases _before_ I open a PR for review next time 😅